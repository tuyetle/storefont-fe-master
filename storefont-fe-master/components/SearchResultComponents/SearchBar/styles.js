import variables from '~/styles/variables';
import themes from '~/styles/themes';
import { css } from 'glamor';

export default{
  searchBar: css({
    margin: '0.25em 1em',
    padding: '0.2em 0',
    boxShadow: variables.$boxShadoSearchBar,
    color: variables.$colorWelcome,
    fontSize: '0.9375em',
    ...themes['font-montserrat'],

    '@media (min-width: 768px)': {
      padding: '0.2em 1em',
    },
    '& .Select-placeholder': {
      fontSize: '0.8875em',
      fontStyle: 'italic',
      ...themes['font-montserrat'],
    },
    '& .Select': {
      '&:focus': {
        outline: 'none',
      },
      '& .Select-clear': {
        display: 'none',
      },
      '&.is-focused:not(.is-open)>.Select-control': {
        boxShadow: 'none',
        outline: 'none',
      },
    },
    '& .Select .Select-control': {
      marginLeft: '0',
      '@media (min-width: 768px)': {
        marginRight: '1em',
      },
    },
    '& .Select--multi .Select-value': {
      boxShadow: variables.$boxShadowTag,
      background: 'none',
      border: 'none',
      color: variables.$colorWelcome,
      borderRadius: '3.125em',
    },
    '& .Select-value-icon': {
      float: 'right',
      border: 'none',
    },
    '& .Select--multi .Select-value-icon': {
      padding: '0.1875em 0.3125em 0.1875em',
      border: 'none',
    },

  }),
  alignCenter: css({
    alignItems: 'center',
    margin: 'auto',
  }),
  saveSearch: css({
    backgroundColor: 'transparent',
    backgroundRepeat: 'no-repeat',
    border: '0',
    cursor: 'pointer',
    overflow: 'hidden',
    outline: 'none',
    margin: '1em',
    color: variables.$locationTextColor,
    fontSize: '0.75em',
    display: 'inline-block',

    '&:before': {
      marginRight: '.5em',
      fontFamily: 'FontAwesome',
      content: '\\f006',
      display: 'none',
    },
    '&:focus': {
      outline: 'none',
    },
  }),
  deleteSearch: css({
    backgroundColor: 'transparent',
    backgroundRepeat: 'no-repeat',
    border: '0',
    cursor: 'pointer',
    overflow: 'hidden',
    outline: 'none',
    marginRight: '1em',
    padding: '0',
    color: variables.$locationTextColor,
    fontSize: '0.75em',
    display: 'inline-block',

    '&:before': {
      marginRight: '.5em',
      fontFamily: 'FontAwesome',
      content: '\\f006',
      display: 'none',
    },
    '&:focus': {
      outline: 'none',
    },
  }),
  success: css({
    border: '1px solid red',
    color: 'white',
    backgroundColor: 'red',
  }),
  filterGroup: css({
    float: 'right',
    textAlign: 'right',
    alignSelf: 'center',
  }),
  listDropdown: css({
    borderLeft: 'none',
    paddingLeft: '1.5em',
    height: '41px',
    lineHeight: '2.5em',
    display: 'inline-block',

    '@media (min-width: 768px)': {
      borderLeft: `1px solid ${variables.$borderdropdown}`,
    },
  }),
  filterArea: css({
    '& svg': {
      verticalAlign: 'middle',
      marginRight: '0',
      '@media (min-width: 768px)': {
        marginRight: '0.4em',
      },
    },
  }),
  groupSell: css({
    display: 'inline-block',
    color: variables.$colorWelcome,
    borderLeft: `1px solid ${variables.$borderdropdown}`,
    padding: '0 0.5em',
    height: '41px',
    lineHeight: '2.5em',
    '& .ant-radio-group-large .ant-radio-button-wrapper': {
      height: '29px',
      lineHeight: '26px',
      '&:before': {
        display: 'none',
      },
    },
    '& .ant-radio-button-wrapper': {
      color: 'black',
      border: '.125em solid transparent',
      textTransform: 'uppercase',
      boxShadow: 'none',
      outline: '0',
      borderRadius: '0',
      fontSize: '0.9375em',

      '&:hover': {
        color: variables.$colorWelcome,
        boxShadow: 'none',
      },

      '&:focus, &:active': {
        borderRadius: '0',
        color: variables.$colorWelcome,
        border: `.125em solid ${variables.$colorHambuger}`,
        boxShadow: 'none',
        outline: 'none',
      },

      '&.ant-radio-button-wrapper-checked': {
        borderRadius: '0',
        border: `.125em solid ${variables.$colorHambuger}`,
        color: variables.$colorWelcome,
        '&:hover': {
          boxShadow: 'none',
        },

        '&:first-child': {
          border: `.125em solid ${variables.$colorHambuger}`,
          borderRadius: '0',
        },
        '&:last-child': {
          border: `.125em solid ${variables.$colorHambuger}`,
          borderRadius: '0',
        },
      },
    },
  }),

  otherFilters: css({
  }),
  filters: css({
    position: 'fixed',
    top: '0',
    right: '0',
    left: '0',
    borderRadius: '0',
    width: '100vw',
    maxWidth: variables.$maxWidthFilter,
    overflow: 'hidden',
    margin: '0',
    backgroundColor: 'white',
    boxShadow: variables.$boxShadowFilter,
    border: `1px solid ${variables.$borderFilter}`,
    color: variables.$colorWelcome,
    padding: '1.5em',
    display: 'none',

    '.closedFilter &': {
      display: 'block',
      zIndex: '9',
      height: '100vh',
      maxHeight: '100vh',
      overflowY: 'scroll',
      overflowX: 'hidden',
      padding: '0',
    },

    '@media (min-width: 768px)': {
      position: 'fixed',
      width: '596px',
      right: '0',
      left: '-18em',
      display: 'block',
    },
  }),
  iconFilters: css({
    '@media (max-width: 767px)': {
      width: '23px',
      height: '20px',
      display: 'block',
      float: 'right',
    },
  }),
};
