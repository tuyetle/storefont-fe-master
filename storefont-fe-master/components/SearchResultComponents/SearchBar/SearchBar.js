import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import { isEmpty, isFunction } from 'lodash';
import { Form, Field, NestedField } from 'react-form';
import { translate } from 'react-i18next';
import classNames from 'classnames';

import SVGFillterIcon from '~/static/assets/img/utilityIcons/ic-fillter.svg';
import FillterIconMobile from '~/static/assets/img/ic-filter-mobile.png';
import { bedOptions, bathroomOptions } from '~/config/searchBar';
import DropdownButton from '~/components/common/DropdownButton/DropdownButton';
import Button from '~/components/common/Button/Button';
import { Mobile, NotMobile } from '~/config/media';

import PriceRange from './PriceRange/PriceRange';
import AreaRange from './AreaRange/AreaRange';
import TransactionTypeFilter from './TransactionTypeFilter/TransactionTypeFilter';
import SearchingTagComponent from './SearchingTagComponent/SearchingTagComponent';
import BedFilters from './BedFilters/BedFilters';
import BathroomFilters from './BathroomFilters/BathroomFilters';
import TheOtherFilters from './TheOtherFilters/TheOtherFilters';
import MobileSearchBar from './MobileSearchBar/MobileSearchBar';

import styles from './styles';

class SearchBar extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { closedFilter: true };
  }

  searchSubmit = (formState) => {
    const { values, touched } = formState;
    if (isEmpty(formState.values)) {
      return;
    }

    if (touched && touched.priceRange
      && touched.priceRange.fromPrice
      && touched.priceRange.toPrice
      && values.priceRange.fromPrice > values.priceRange.toPrice) {
      return;
    }

    if (touched && touched.areaRange
      && touched.areaRange.fromArea
      && touched.areaRange.toArea
      && values.areaRange.fromArea > values.areaRange.toArea) {
      return;
    }

    if (isFunction(this.props.changeURL)) this.props.changeURL(values);

    this.props.searchListing(values);
  }

  saveSearchAlert = () => {
    this.props.saveSearch({ url: 'mockDick' });
  }

  toggleFilter = (evt) => {
    evt.preventDefault();
    this.setState({ closedFilter: !this.state.closedFilter });
  }

  renderOther = () => {
    if (typeof window === 'undefined') return null;

    const {
      t, suggestions,
      selectedFilter: {
        priceRange,
        beds,
        bathrooms,
        areaRange,
        otherFilters,
      },
      getSuggestions, selectedSuggestion,
    } = this.props;

    const listingTypeOptions = [{
      value: 'buy',
      label: t('common:buy'),
    },
    {
      value: 'loan',
      label: t('common:loan'),
    }];

    return (
      <NotMobile>
        <Form onChange={this.searchSubmit}>
          {
            () => (
              <div className="container-fluid">
                <form
                  id="searchListingForm"
                  className={`row ${styles.searchBar}`}
                >
                  <div className="col-4 col-md-3 col-lg-4">
                    <Field
                      field="searchKeyworks"
                      component={SearchingTagComponent}
                      promptText={t('searchingWith')}
                      noResultsText={t('noResultsText')}
                      placeholder={t('searchingGuideline')}
                      suggestions={suggestions}
                      selectedSuggestion={selectedSuggestion}
                      getSuggestions={getSuggestions}
                    />
                  </div>
                  <div className={`col-8 col-md-9 col-lg-8 ${styles.filterGroup}`}>
                    <Button className={`${styles.saveSearch} ${this.props.saveSearchResult.isSuccess ? styles.success : ''}`} label={t('common:saveSearch')} onClickHandler={this.saveSearchAlert} />
                    <Button className={`${styles.deleteSearch}`} label={t('common:deleteSearch')} />
                    <div className={`${styles.groupSell}`}>
                      <Field
                        field="transactionType"
                        component={TransactionTypeFilter}
                        defaultValue={listingTypeOptions[0].value}
                        options={listingTypeOptions}
                        className={`${styles.transactionType}`}
                      />
                    </div>
                    <div className={`${styles.listDropdown}`}>
                      <div className="row">
                        <section name="search-price">
                          <DropdownButton label={t('common:price')}>
                            <NestedField
                              field="priceRange"
                              component={PriceRange}
                              millionLabelForUnit={t('common:million')}
                              billionLabelForUnit={t('common:billion')}
                              fromPriceLabel={t('common:priceLowest')}
                              toPriceLabel={t('common:priceHighest')}
                              fromPriceDefaultValue={
                                priceRange && priceRange.fromPrice
                              }
                              toPriceDefaultValue={
                                priceRange && priceRange.toPrice
                              }
                              t={t}
                            />
                          </DropdownButton>
                        </section>
                        <section name="search-bedrooms">
                          <DropdownButton label={t('common:bedrooms')}>
                            <Field
                              field="beds"
                              component={BedFilters}
                              bedOptionsConfig={bedOptions}
                              defaultValue={beds || []}
                            />
                          </DropdownButton>
                        </section>
                        <section name="search-bathrooms">
                          <DropdownButton label={t('common:bathrooms')}>
                            <Field
                              field="bathrooms"
                              component={BathroomFilters}
                              bathroomOptions={bathroomOptions}
                              defaultValue={bathrooms || []}
                            />
                          </DropdownButton>
                        </section>
                        <section name="area-search">
                          <DropdownButton label={t('common:area')}>
                            <NestedField
                              field="areaRange"
                              component={AreaRange}
                              fromAreaLabel={t('common:from')}
                              toAreaLabel={t('common:to')}
                              formAreaDefaultValue={
                                areaRange && areaRange.fromArea
                              }
                              toAreaDefaultValue={
                                areaRange && areaRange.toArea
                              }
                              t={t}
                            />
                          </DropdownButton>
                        </section>
                        <DropdownButton
                          className={`${styles.otherFilters}`}
                          label={<div className={`${styles.filterArea}`}><SVGFillterIcon /> {t('common:filters')}</div>}
                          showCaret={false}
                        >
                          <div className="container">
                            <div className={`${styles.filters}`}>
                              <Field
                                field="otherFilters"
                                component={TheOtherFilters}
                                categories={this.props.categories}
                                furnitures={this.props.furnitures}
                                utilities={this.props.utilities}
                                defaultValue={otherFilters || {}}
                                t={t}
                              />
                            </div>
                          </div>
                        </DropdownButton>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            )
          }
        </Form>
      </NotMobile>
    );
  }

  renderMobile = () => {
    if (typeof window === 'undefined') return null;

    const {
      t, suggestions,
      getSuggestions, selectedSuggestion, selectedFilter,
    } = this.props;
    const otherFilters = selectedFilter && selectedFilter.otherFilters;
    const { closedFilter } = this.state;

    return (
      <Mobile>
        <Form onChange={this.searchSubmit}>
          {
            () => (
              <div className="container-fluid">
                <form
                  id="searchListingMobileForm"
                  className={`row ${styles.searchBar}`}
                >
                  <div className="col-10">
                    <SearchingTagComponent
                      field="searchKeyworks"
                      promptText={t('searchingWith')}
                      noResultsText={t('noResultsText')}
                      placeholder={t('searchingGuideline')}
                      suggestions={suggestions}
                      selectedSuggestion={selectedSuggestion}
                      getSuggestions={getSuggestions}
                    />
                  </div>
                  <div className={`col-2 ${styles.filterGroup}`}>
                    <div
                      className={classNames(`${styles.otherFilters}`, closedFilter ? '' : 'closedFilter')}
                    >
                      <input type="image" className={`${styles.iconFilters}`} onClick={this.toggleFilter} src={FillterIconMobile} alt="" />
                      <MobileSearchBar
                        selectedValues={otherFilters}
                        categories={this.props.categories}
                        furnitures={this.props.furnitures}
                        utilities={this.props.utilities}
                        closeFilter={this.toggleFilter}
                        t={t}
                        defaultValues={selectedFilter}
                      />
                    </div>
                  </div>
                </form>
              </div>
            )
          }
        </Form>
      </Mobile>
    );
  }

  render() {
    return (
      <Fragment>
        {
          this.renderMobile()
        }
        {
          this.renderOther()
        }
      </Fragment>
    );
  }
}

SearchBar.propTypes = {
  categories: PropTypes.array.isRequired,
  furnitures: PropTypes.array.isRequired,
  utilities: PropTypes.array.isRequired,
  getSuggestions: PropTypes.func.isRequired,
  selectedSuggestion: PropTypes.object,
  changeURL: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  suggestions: PropTypes.array,
  searchListing: PropTypes.func.isRequired,
  saveSearch: PropTypes.func.isRequired,
  saveSearchResult: PropTypes.object,
  selectedFilter: PropTypes.object,
};

SearchBar.defaultProps = {
  suggestions: [],
  selectedSuggestion: undefined,
  selectedFilter: {},
  saveSearchResult: {},
};

export default translate(['searchResult', 'common'])(SearchBar);
