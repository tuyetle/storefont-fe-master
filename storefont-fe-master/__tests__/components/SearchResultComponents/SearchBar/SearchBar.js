import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '~/components/SearchResultComponents/SearchBar/SearchBar';

describe('(Component) SearchBar', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      categories: [{
        name: 'House',
        value: '1',
      }],
      furnitures: [{
        name: 'Sofa',
        value: '2',
      }],
      utilities: [{
        name: 'Wifi',
        value: '3',
      }],
      t() { },
      searchListing: () => {},
      saveSearch: () => {},
      getSuggestions: () => {},
      changeURL: () => { },
      suggestions: [],
      selectedFilter: {
        transactionType: 'sell',
        priceRange: {
          fromPrice: 0,
          toPrice: 0,
        },
        areaRange: {
          fromArea: 0,
          toArea: 0,
        },
        bathrooms: [],
        beds: [],
        otherFilters: {},
      },
    };

    component = <SearchBar {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

