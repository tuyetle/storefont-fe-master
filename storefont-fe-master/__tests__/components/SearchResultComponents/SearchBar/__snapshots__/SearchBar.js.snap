// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) SearchBar should render conrectly 1`] = `null`;
